astropy/visualization/wcsaxes/coordinate_helpers.py

```diff
@@ -716,7 +716,7 @@ def set_ticklabel_visible(self, visible):
         """
         self._ticklabels.set_visible(visible)
 
-    def set_axislabel(self, text, minpad=1, **kwargs):
+    def set_axislabel(self, text, minpad=1, loc="center", **kwargs):
```

Comment:
here too, this should never be allowed positionally

```suggestion
    def set_axislabel(self, text, minpad=1, *, loc="center", **kwargs):
```